### ISSUE TITLE
[aria-label] does not update when dynamic field label changes

### ISSUE THREAD

**Issue by mpiasta-ca** [angular/material#8495]
**Actual behavior**:
-  What is the issue? When `<md-input-container ...` uses a dynamic `<label>`, when that label changes, the `aria-label` for the `<md-select>` field is not updated to reflect the change.
-  What is the expected behavior? `<label>` should be watched so that a change to it will be reflected in `aria-label`. 

**CodePen** or Steps to reproduce the issue: *
-  CodePen Demo which shows your issue : http://codepen.io/anon/pen/aNrgyV?editors=1010
-  Details: Wait 3 seconds to see the `<label>` on the `<md-select>` change from "Minutes" to "Hours". For `<md-select>` this change won't appear because the `<md-select-value>` element does not update either. You have to expand then collapse the options list to see it reflect the new label. But if you inspect the `<md-select>` you will see the aria-label still reflects the old label value of "Minutes" (should be "Hours" now).

**Angular Versions**: *
-  Angular Version: 1.5
-  Angular Material Version: 1.0.8...? (used codepen template)

**Additional Information**:
-  Browser Type: *all
-  Browser Version: *all
-  OS: *all
-  Stack Traces: n/a

**Comment by ThomasBurleson** [comment]
This issue is closed as part of our ‘Surge Focus on Material 2' efforts.
      For details, see our forum posting @ http://bit.ly/1UhZyWs.
